darwin/client.py

```diff
@@ -86,15 +89,26 @@ def list_remote_datasets(self, team_slug: Optional[str] = None) -> Iterator[Remo
         response: List[Dict[str, Any]] = cast(List[Dict[str, Any]], self._get("/datasets/", team_slug=team_slug))
 
         for dataset in response:
-            yield RemoteDataset(
-                name=dataset["name"],
-                slug=dataset["slug"],
-                team=team_slug or self.default_team,
-                dataset_id=dataset["id"],
-                item_count=dataset["num_images"] + dataset["num_videos"],
-                progress=dataset["progress"],
-                client=self,
-            )
+            if dataset.get("version", 1) == 2:
+                yield RemoteDatasetV2(
+                    name=dataset["name"],
+                    slug=dataset["slug"],
+                    team=team_slug or self.default_team,
+                    dataset_id=dataset["id"],
+                    item_count=dataset.get("item_count", dataset["num_images"] + dataset["num_videos"]),
+                    progress=dataset["progress"],
+                    client=self,
+                )
+            else:
+                yield RemoteDatasetV1(
+                    name=dataset["name"],
+                    slug=dataset["slug"],
+                    team=team_slug or self.default_team,
+                    dataset_id=dataset["id"],
+                    item_count=dataset["num_images"] + dataset["num_videos"],
+                    progress=dataset["progress"],
+                    client=self,
+                )
```

Comment:
Minor, but since we are almost repeating this resolution logic for remote dataset class, we could think of creating a `resolve_remote_dataset` function.

Reply:
since it's limited to only be inside client.py I'm not too keen on trying it to DRY it. I think that they will diverge more as time goes on.